spring-cloud-kubernetes-discovery/src/main/java/org/springframework/cloud/kubernetes/discovery/KubernetesDiscoveryClient.java

```diff
@@ -68,8 +70,9 @@ public String description() {
 
 	@Override
 	public List<ServiceInstance> getInstances(String serviceId) {
-		KubernetesServiceInstance[] responseBody = rest
-				.getForEntity(discoveryServerUrl + "/apps/" + serviceId, KubernetesServiceInstance[].class).getBody();
+		DefaultKubernetesServiceInstance[] responseBody = rest
```

Comment:
we don't pase into the custom `KubernetesServiceInstance` anymore, but as all other discovery clients do, into the  `DefaultKubernetesServiceInstance`. This also makes sure that we align all 3 discovery implementations.